home/tests/test_api.py

```diff
@@ -991,15 +1010,76 @@ def test_whatsapp_template_fields(self, uclient: Any) -> None:
         page = self.create_content_page(
             whatsapp_template_category=ContentPage.WhatsAppTemplateCategory.MARKETING,
             whatsapp_template_slug="test_template",
+            whatsapp_template_submission_name="test_template_1",
         )
 
         response = uclient.get(f"/api/v2/pages/{page.id}/?whatsapp")
         body = response.json()["body"]
         assert body["is_whatsapp_template"]
-        assert body["whatsapp_template_name"] == "test_template"
+        assert body["whatsapp_template_name"] == "test_template_1"
         assert body["text"]["value"]["message"] == "Test WhatsApp Template Message 1"
         assert body["whatsapp_template_category"] == "MARKETING"
 
+    def test_whatsapp_template_via_v2_api(self, uclient: Any) -> None:
+        """
+        Check that we can see the appropriate data in the V2 API detail view when using a Whatsapp Template
+        """
+        page = self.create_content_page(
+            whatsapp_template_category=ContentPage.WhatsAppTemplateCategory.MARKETING,
+            whatsapp_template_slug="test_template",
+            whatsapp_template_image=1,
+            whatsapp_template_example_values=[
+                ("example_values", "Ev1"),
+                ("example_values", "Ev1"),
+            ],
+            whatsapp_template_submission_name="test_template_1",
+            buttons=[{"type": "next_message", "value": {"title": "Tell me more"}}],
```

Comment:
Do we need a similar test with list options instead of buttons? (Similarly for the non-template version, and also in the v3 api.)